qt/aqt/imageocclusion.py

```diff
@@ -0,0 +1,62 @@
+# Copyright: Ankitects Pty Ltd and contributors
+# License: GNU AGPL, version 3 or later; http://www.gnu.org/licenses/agpl.html
+
+from __future__ import annotations
+
+import aqt
+import aqt.deckconf
+import aqt.main
+import aqt.operations
+from aqt.qt import *
+from aqt.utils import addCloseShortcut, disable_help_button, restoreGeom, saveGeom, tr
+from aqt.webview import AnkiWebView
+
+
+class ImageOcclusionDialog(QDialog):
+    silentlyClose = True
+
+    def __init__(
+        self,
+        mw: aqt.main.AnkiQt,
+        mode: str,
+        path=None,
+        notetype_id=None,
+        note_id=None,
+    ) -> None:
+        QDialog.__init__(self, mw)
+        self.mw = mw
+        self._setup_ui(mode, path, notetype_id, note_id)
+        self.show()
+
+    def _setup_ui(
+        self, mode: str, path: str | None, notetype_id: int, note_id: int | None
+    ) -> None:
+        self.TITLE = tr.notetypes_image_occlusion_name()
+        self.setWindowModality(Qt.WindowModality.ApplicationModal)
+        self.mw.garbage_collect_on_dialog_finish(self)
+        self.setMinimumSize(400, 300)
+        disable_help_button(self)
+        restoreGeom(self, self.TITLE)
+        addCloseShortcut(self)
+
+        self.web = AnkiWebView(title=self.TITLE)
+        self.web.setVisible(False)
+        self.web.load_ts_page("image-occlusion")
+        layout = QVBoxLayout()
+        layout.setContentsMargins(0, 0, 0, 0)
+        layout.addWidget(self.web)
+        self.setLayout(layout)
+
+        if mode == "add":
+            options = {"kind": mode, "imagePath": path, "notetypeId": notetype_id}
+        elif mode == "edit":
+            options = {"kind": mode, "noteId": note_id}
+
+        self.web.eval(f"anki.setupImageOcclusion({options});")
```

Comment:
Adding an explicit json.dumps() call here would be better than relying on Python's string repr happening to be valid JS.

Reply:
I have changed the implementation to include setupImageOcclusion method in note editor.